### PR TITLE
Resolve 'bytes'/'binary' discrepancy

### PR DESCRIPTION
**Pull request by popematt**
**Issue #, if available:**

Follow up to #126 

**Description of changes:**

* Removed the use of `bytes` as a fragment keyword in favor of `binary`
* Text fragments can now accept a mix of string literals a integers (which are interpreted as bytes)
* I chose not to use `bytes` to be a general purpose input fragment because if we're combining fragments, we need to know what the intended format is to be. If `bytes` could represent binary or text, then something like `(ivm 1 1) (bytes ...)` would be ambiguous w.r.t. whether the IVM fragment should be converted to `E0 01 01 EA` or `$ion_1_1`.

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: .github/workflows/main.yaml
Comment (zslayton) [inline .github/workflows/main.yaml:11]: Why is this no longer needed?

**Review by zslayton — APPROVED** [review]

**Review by tgregg — APPROVED** [review]

**Comment by popematt** [comment]
@toddjonker, I've got two approvals. I'm going to merge this, but if you have any concerns we can revisit this.

**Pull request merged**
